### PR TITLE
feat: add a bunch of explanation, as well as …

### PR DESCRIPTION
**Pull request by todpunk**
a placeholder for not-yet-ready tutorial

### CODE REVIEW

File: explanation/capacity.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+## Monitoring Usage
+
+The most important part of capacity planning is that it is done continuously. There is no capacity today that will not be wrong tomorrow. Luckily, this is enabled very easily by what we should be doing already, monitoring!
+
+Keep a track record of metrics on a variety of pieces of Kafka's performance. Keep track of lagging Consumer groups, Partitions, ISRs, storage capacity, CPU and Memory usage, and anything else you can measure. If using Prometheus metrics, you want to keep in mind the cardinality of labels you add so you don't lose monitoring capabilities as labels explode, but measure as much as possible within that constraint. Even if something isn't a core metric, it can help paint a picture of why the core metric behaves the way it does. For example, your Consumer lag may happen because open file handles has reached your OS limit, so reads are having contention on file handles but you won't see that in CPU or Memory stats, and not likely on storage capacity.
```
Comment (codelite7) [inline explanation/capacity.md]: Maybe add links to some docs on prometheus kafka and grafana dashboards? Something like https://computingforgeeks.com/monitor-apache-kafka-with-prometheus-and-grafana/

File: explanation/data-loss.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+Cloud Native thinking aspires to treat as many things as ephemeral as possible. Chaos Engineering is a way to test a lot of the fault tolerant assumptions we make. Catalyst Squad likes to ask this a simple way: can I delete this?
+
+If the answer is yes because the system will continue running and the service is still available, the system is fault tolerant to that item being deleted. If data loss will occur if I delete this, it is not fault tolerant.
+
+Kafka's workhorse concept is the Partition. An In Sync Replica (ISR) is the copy of that Partition's data, where it could resume the responsibilities of the Partition leader and be the actual Partition instead of an ISR.
+
+You can delete an ISR's files off disk, but Kafka will have errors and not know what to do. Deleting an entire node will only result in failures of any Partitions that don't have any ISRs left. This is the critical point. ISRs must exist for fault tolerance and recovery. If Kafka can't figure out how to have the minimum required ISRs, it can't accept new messages. It won't lost data, but Producers might. Now we have to think about Producers in addition to Kafka. Well, one thing at a time.
```
Comment (benjyiw) [inline explanation/data-loss.md]: > It won't lost data, but Producers might.
lost -> lose?

Comment (codelite7) [inline explanation/data-loss.md]: Change "It won't lost data," to "It won't lose data"

File: explanation/data-loss.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+## What about dead letter queues?
+
+It is not an uncommon request to get Kafka users asking about dead letter queues. This is a good sign of two things. First, Kafka is being used as a queue, which is very bad. Kafka is a parallelized append-only log, not a work queue. Nothing can change this. Consumer offsets can get reset in a variety of scenarios. Will items in the topic be reprocessed or do they expect at-most-once semantics? Second, this problem never ends, so this is a time for conversation about properly engineering a Kafka use case.
```
Comment (benjyiw) [inline explanation/data-loss.md]: Using
> This is a good sign of two things

and then going on to talk about problems had me confused a little. maybe something like "This is an indicator of two things" ?

File: explanation/data-loss.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+If a dead letter queue is requested, what happens when the error happens on the dead letter queue Consumer? Does progress halt there? Do we need a dead letter queue for the dead letter queue? We think the answer is obviously no, but we've also seen this done more than once by very different groups.
+
+Our advice is to never dead letter queue, and if you need to reprocess something, capture the errors in a different context. This could even look very much like a dead letter queue, but it should in actuality be more like a Sentry style capturing mechanism where errors can be handled out of band. This means that the ability to handle a message days late all while handling all the other messages live. Some will say this isn't tenable, but this is necessary with a dead letter queue as well.
```
Comment (benjyiw) [inline explanation/data-loss.md]: > This means that the ability to handle a message days late all while handling all the other messages live.

A little confusing to me, maybe "This means that you gain the ability to handle a message days late" ?

File: explanation/multi-cluster.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+A lot of users wish to have region-redundant data in Kafka. We should note here that most users don't need this, it just sounds like a good idea. Most teams will have much more important problems should an entire cloud region go down. Be very careful if choosing Kafka for a use case where multiple regions is a very real requirement, for backups or otherwise.
+
+Here we will discuss the extra trade-offs that come with moving into multiple Kafka clusters coordinating.
+
+## There is no Multi-Cluster
+
+It's important to understand that all multiple cluster kafka deployments are in a very real way simply single clusters glued together in a way that they are unaware of each other.
+
+Mirror Maker and Mirror Maker 2 can both make data redundant on multiple clusters. It does this, in all cases, by having a normal Consumer client read from TopicA in ClusterA, and Produce to TopicA in ClusterB. Just because they are both named TopicA does not mean they are the same topic. There's no guarantees about ordering data (you can arrange it so that this is largely preserved, but this is precarious).
```
Comment (codelite7) [inline explanation/multi-cluster.md]: Maybe add links to mirror maker

**Review by codelite7 — DISMISSED** [review]

**Review by benjyiw — DISMISSED** [review]

File: explanation/data-loss.md (hunk 2 above)
Comment (benjyiw) [reply explanation/data-loss.md]: lol zack pointed out the same thing

**Review by codelite7 — APPROVED** [review]

**Pull request merged**